src/use_searcher.rs

```diff
@@ -1,22 +1,16 @@
-use std::{
-    cell::{OnceCell, RefCell},
-    ptr,
-    rc::Rc,
-};
+use std::{cell::RefCell, collections::HashMap, rc::Rc};
 
 use crate::{searcher::Searcher, Args};
 
 thread_local! {
-    static SEARCHER: OnceCell<(Rc<RefCell<Searcher>>, *const Args)> = Default::default();
+    static SEARCHER_PER_ARGS_INSTANCE: RefCell<HashMap<*const Args, Rc<RefCell<Searcher>>>> = Default::default();
```

Comment:
I'd forgotten that we have now invalidated this invariant (but glad I was asserting it!) (that there's only ever once instance of `Args` used during a program execution)

So just made this a per-`Args`-instance map instead (yes in theory then you should probably be more "legitimately cache-like" and eg evict once you get past a certain size but not worrying about that for now)

This same change should be made to the printer which is using the same pattern